app/views/unattended/provisioning_templates/snippet/kickstart_kernel_options.erb

```diff
@@ -39,19 +39,19 @@ description: |
   # both current and legacy syntax provided
   if (is_fedora && os_major >= 33) || (rhel_compatible && os_major >= 9)
     if subnet4 && !subnet4.dhcp_boot_mode?
-      options.push("inst.ks=#{foreman_url('provision', static: '1')}")
+      options.push("inst.ks=#{foreman_url('provision', static: '1').gsub('&','\\\\&')}")
```

Comment:
```suggestion
      options.push("inst.ks=" + format("'%s'", foreman_url('provision', static: '1')))
```

one thought: quoting the URL for grub should also work, could prevent similar future issues from occurring for other characters than `&`, and might be a bit easier to read for the next engineer vs. trying to figure out which escapes are in which layer... what do you think?

either way, it's probably good to to give the same treatment as well to the lines setting `inst.repo` and `inst.stage2`